drivers/hx711/hx711.c

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2019 Philipp-Alexander Blum <philipp-blum@jakiku.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_hx711 HX711 digital scale sensor
+ * @ingroup     drivers_sensors
+ * @brief       Driver for the HX711 digital scale sensor
+ *
+ * @{
+ * @file
+ * @brief       HX711 driver
+ *
+ * @author      Philipp-Alexander Blum <philipp-blum@jakiku.de>
+ */
+
+#include "periph/gpio.h"
+#include "periph/gpio_util.h"
+#include "xtimer.h"
+
+#include "hx711.h"
+#include "hx711_params.h"
+
+/**
+ * @brief Needs to wait for device to be ready while initialization
+ */
+void wait_for_ready(void)
+{
+    while(gpio_read(HX711_PARAM_DOUT) > 0){
+        xtimer_usleep(HX711_PARAM_SLEEP_TIME);
+    }
+}
+
+/**
+ * @brief Gets a single 24-Bit value from the HX711.
+ * @return The 24-Bit shift in value
+ */
+int32_t hx711_read(void)
```

Comment:
Are `hx711_read`, `hx711_read_average`, `hx711_get_value` supposed to be private functions? If so, they should be named like `_read` and declared static. The same for `wait_for_ready`. If, on the other hand, they are part of the public API then please add them to the header and move the documentation there.